connector/azure-apimanager/src/test/java/it/pagopa/selfcare/pagopa/backoffice/connector/azure_apim/AzureApiManagerClientTest.java

```diff
@@ -74,10 +118,77 @@ void createInstitutionSubscription_ok() throws NoSuchFieldException, IllegalAcce
+        //given
+
+        SubscriptionsImpl subscriptionMock = mock(SubscriptionsImpl.class);////////////////////////////
+        UserSubscriptionsImpl userSubscriptionsMock = mock(UserSubscriptionsImpl.class);
+        AzureApiManagerClient managerClient = new AzureApiManagerClient(serviceName,
+                resourceGroup,
+                subscriptionId,
+                tenantId);
+        SubscriptionContract contractMock = mock(SubscriptionContract.class);
+        //DummyKeyContract keyContract = mockInstance(new DummyKeyContract());
+        // PagedIterable<SubscriptionContract> subscriptionContractList = getPi(1);
+        PagedIterable<SubscriptionContract> subscriptionContractListMock = mockInstance(getPi(1));
+
+        InstitutionApiKeys apiKeysMock = TestUtils.mockInstance(new InstitutionApiKeys(), "sePrimaryKey", "setSecondaryKey", "setDisplayName");
+        apiKeysMock.setPrimaryKey("primaryKey");
+        apiKeysMock.setSecondaryKey("secondaryKey");
+        apiKeysMock.setDisplayName("displayName");
+        //InstitutionApiKeys apiKeysMock = mockInstance(new InstitutionApiKeys());
+        List<InstitutionApiKeys> apiKeysMockList = mockInstance(List.of(apiKeysMock));
+        ApiManagementManager managerMock = mock(ApiManagementManager.class);
+        AzureApiManagerClient apiManagerClient = mock(AzureApiManagerClient.class);
+        mockAzureApiManagement(managerClient, managerMock);
+
+        when(managerMock.subscriptions())
+                .thenReturn(subscriptionMock);
+        when(managerMock.subscriptions().createOrUpdate(any(), any(), any(), any()))
+                .thenReturn(contractMock);
+        when(apiManagerClient.getApiSubscriptions(any()))
+                .thenReturn(apiKeysMockList);
+        when(managerMock.userSubscriptions())
+                .thenReturn(userSubscriptionsMock);
+        when(managerMock.userSubscriptions().list(any(), any(), any()))
+                .thenReturn(subscriptionContractListMock);
+
+//        when(managerMock.userSubscriptions().list(any(), any(), any()).stream().map())
+//                .thenReturn(subscriptionContractListMock);
+
+//        when(managerMock.subscriptions().listSecretsWithResponse(any(), any(), any(), any()))
+//                .thenReturn(keyContract);
+//        when(apiManagerClient.createInstitutionSubscription(any(), any()))
+//                .thenReturn(apiKeysMockList);
+
+        //when
+        managerClient.createInstitutionSubscription(institutionId, institutionName, scope, subscriptionId, serviceName);
+        //then
+//        assertNotNull(apiKeys);
+//        assertNotNull(apiKeys.get(0).getPrimaryKey());
+//        assertNotNull(apiKeys.get(0).getSecondaryKey());
+//        assertEquals(keyContract.getValue().primaryKey(), apiKeys.get(0).getPrimaryKey());
+//        assertEquals(keyContract.getValue().secondaryKey(), apiKeys.get(0).getSecondaryKey());
```

Comment:
why are these commented?

Reply:
cause the method is void and there are no fields to check